### PR TITLE
use qdrant client v1.7

### PR DESCRIPTION
**Pull request by coszio**
Updates the qdrant client to 1.7 along with its needed refactors

### CODE REVIEW

**Comment by Anush008** [comment]
who-me

**Review by Anush008 — APPROVED** [review]
🫡

File: src/search.rs [hunk 1]
```diff
@@ -66,6 +66,9 @@ impl SearchProcessor {
             vector_name,
             with_vectors: None,
             read_consistency: self.args.read_consistency.map(Into::into),
+            shard_key_selector: None,
+            timeout: self.args.timeout.map(|v| v as u64),
```
Comment (KShivendu) [inline src/search.rs:70]: This `timeout` part could be a good addition.

**Review by KShivendu — CHANGES_REQUESTED** [review]
We already did these client upgrades in the `dev` branch. Please use that branch instead.

**Review by timvisee — CHANGES_REQUESTED** [review]
Thanks!

Though, our `master` and `dev` have diverged a bit. Instead of changing the version here on master directly, I think it would be better to rely on all the changes we made in dev instead.

We probably have to:
- cherry-pick dev changes into master
- update version in master

I'll make sure to arrange this with @KShivendu.

Drafting this now, since this will likely be obsolete after the above mentioned change. We can always revive this later if our efforts fail.

**Comment by timvisee** [comment]
Our `master` is up to date now and uses Qdrant 1.7. Closing this now.

**Comment by coszio** [comment]
Oops, didn't notice we had a dev branch here too. Thanks for the correction

**Pull request closed without merging**